File: frontend/dto/stats/homeStatsResponse.ts
import * as z from "zod";
import { RockItSongWithAlbumResponse } from "@/dto/rockItSongWithAlbumResponse";

export const HomeStatsResponse = z.object({
    songsByTimePlayed: z.array(RockItSongWithAlbumResponse),
    randomSongsLastMonth: z.array(RockItSongWithAlbumResponse),
    nostalgicMix: z.array(RockItSongWithAlbumResponse),
    hiddenGems: z.array(RockItSongWithAlbumResponse),
    communityTop: z.array(RockItSongWithAlbumResponse),
    monthlyTop: z.array(RockItSongWithAlbumResponse),
    moodSongs: z.array(RockItSongWithAlbumResponse),
});

File: backend/core/responses/homeStatsResponse.py
from typing import List
from pydantic import BaseModel, Field

from backend.core.responses.baseSongWithAlbumResponse import BaseSongWithAlbumResponse


class HomeStatsResponse(BaseModel):
    songsByTimePlayed: List[BaseSongWithAlbumResponse] = Field(default_factory=list)
    randomSongsLastMonth: List[BaseSongWithAlbumResponse] = Field(default_factory=list)
    nostalgicMix: List[BaseSongWithAlbumResponse] = Field(default_factory=list)
    hiddenGems: List[BaseSongWithAlbumResponse] = Field(default_factory=list)
    communityTop: List[BaseSongWithAlbumResponse] = Field(default_factory=list)
    monthlyTop: List[BaseSongWithAlbumResponse] = Field(default_factory=list)
    moodSongs: List[BaseSongWithAlbumResponse] = Field(default_factory=list)
